geomesa-raster/src/main/scala/org/locationtech/geomesa/raster/data/AccumuloRasterQueryPlanner.scala

```diff
@@ -30,31 +30,63 @@ import org.locationtech.geomesa.core.iterators._
 import org.locationtech.geomesa.raster._
 import org.locationtech.geomesa.raster.index.RasterIndexSchema
 import org.locationtech.geomesa.raster.iterators.RasterFilteringIterator
-import org.locationtech.geomesa.utils.geohash.{BoundingBox, GeoHash, GeohashUtils}
+import org.locationtech.geomesa.utils.geohash.{BoundingBox, GeohashUtils}
 import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
 import org.opengis.feature.simple.SimpleFeatureType
 import org.opengis.filter.Filter
 
+import scala.collection.JavaConversions._
+
 // TODO: Constructor needs info to create Row Formatter
 // right now the schema is not used
 // TODO: Consider adding resolutions + extent info  https://geomesa.atlassian.net/browse/GEOMESA-645
 case class AccumuloRasterQueryPlanner(schema: RasterIndexSchema) extends Logging with IndexFilterHelpers {
 
+  // Dotting without the dots.
+  def shorten(set: Seq[String]): Iterator[String] = {
+    val len = set.headOption.map(_.length).getOrElse(0)
+    for {
+      i <- (1 to len-1).iterator
+      hash <- set.map(_.take(i)).distinct
+      newStr = hash.take(i)
+    } yield newStr
+  }
+
   def getQueryPlan(rq: RasterQuery, availableResolutions: List[Double]): QueryPlan = {
 
     // TODO: WCS: Improve this if possible
     // ticket is GEOMESA-560
     // note that this will only go DOWN in GeoHash resolution -- the enumeration will miss any GeoHashes
     // that perfectly match the bbox or ones that fully contain it.
-    val closestAcceptableGeoHash = GeohashUtils.getClosestAcceptableGeoHash(rq.bbox).getOrElse(GeoHash("")).hash
-    val hashes = (BoundingBox.getGeoHashesFromBoundingBox(rq.bbox) :+ closestAcceptableGeoHash).toSet.toList
+    val closestAcceptableGeoHash = GeohashUtils.getClosestAcceptableGeoHash(rq.bbox)
```

Comment:
If this ticket address GM-560, then we can remove the comment above about GM-560;)